c10/CMakeLists.txt

```diff
@@ -44,6 +46,10 @@ if(HAVE_SOVERSION)
   set_target_properties(c10 PROPERTIES
       VERSION ${TORCH_VERSION} SOVERSION ${TORCH_SOVERSION})
 endif()
+if(APPLE AND BUILD_ARM64)
```

Comment:
@malfet Should I have a narrower conditional than APPLE? Is there a more precise way we can tell if the machine is/will be ARM64?

Reply:
This is fine, although `BUILD_UNIVERSAL` is probably a better name for the option. 
P.S. Universal binary approach might not work with PIP, will find it out today once I get M1 Mini.